### PR TITLE
ICDS dashboard refactoring

### PR DESCRIPTION
**Pull request by czue**
Some minor structural changes designed to make it easier to share as much as possible between the current dashboard and an upcoming mobile dashboard. There are no functional changes here, just code structure.

Easiest reviewed by commit.

### CODE REVIEW

File: custom/icds_reports/static/js/icds_dashboard_utils.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* global angular hqImport */
+
+function populateDashboardConstants() {
```
Comment (czue) [inline custom/icds_reports/static/js/icds_dashboard_utils.js]: I'm pretty sure this should be using hqDefine, though I was largely following the pattern used by the rest of the dashboard...

**Review by czue — COMMENTED** [review]
Also pinging @orangejenny for any pointers since this is largely front-end and I wasn't too sure if I was doing it right.

File: custom/icds_reports/static/js/icds_dashboard_utils.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/* global angular hqImport */
+
+window.populateDashboardConstants = function populateDashboardConstants() {
```
Comment (czue) [inline custom/icds_reports/static/js/icds_dashboard_utils.js]: I'm pretty sure this should be using hqDefine, though I was largely following the pattern used by the rest of the dashboard...

Comment (orangejenny) [reply custom/icds_reports/static/js/icds_dashboard_utils.js]: Yeah, the more modular way to do this is to wrap this whole file like
```
hqDefine("js/icds_dashboard_utils", function () {
    ...
    return {
        populateDashboardConstants: populateDashboardConstants,
    };
});
```

And then call it with `hqImport("js/icds_dashboard_utils").populateDashboardConstants()`

Comment (czue) [reply custom/icds_reports/static/js/icds_dashboard_utils.js]: thanks, I'll try that.

does compression work properly with hqImport blocks?

**Comment by orangejenny** [comment]
@czue I pretty much ignore the dashboard because it's so out of step with the rest of HQ's front end - it's the only place using angular, one of very few places that doesn't inherit from `hqwebapp/base.html`, and the only place that uses non-trivial inline js. With that in mind, everything you changed/added here seems fine.

I'm curious, is the mobile dashboard going to be angular-based?

File: custom/icds_reports/static/js/icds_dashboard_utils.js (hunk 2 above)
Comment (orangejenny) [reply custom/icds_reports/static/js/icds_dashboard_utils.js]: yeah, compression should be fine

**Comment by czue** [comment]
> I'm curious, is the mobile dashboard going to be angular-based?

Yeah that's the plan. We want to share as much as possible with web dashboard and that's the stack the main devs are most comfortable in.

File: custom/icds_reports/static/js/icds_dashboard_utils.js [hunk 3]
```diff
@@ -1,33 +1,39 @@
-/* global angular hqImport */
+/* global angular hqDefine hqImport */
```
Comment (orangejenny) [inline custom/icds_reports/static/js/icds_dashboard_utils.js]: Nitpick: `hqDefine` and `hqImport` don't need to go in here, they're in the list of blessed globals: https://github.com/dimagi/commcare-hq/blob/e149006864f5473886e9cdf3a29a5798560c5985/.eslintrc.js#L11

Comment (czue) [reply custom/icds_reports/static/js/icds_dashboard_utils.js]: ah good to know, thanks

**Comment by czue** [comment]
@calellowitz @Rohit25negi @orangejenny anyone feel like reviewing/approving this?

**Review by orangejenny — APPROVED** [review]

**Pull request merged**